MergerLogic/Clients/S3Client.cs

```diff
@@ -112,5 +95,14 @@ public void UpdateTile(Tile tile)
                 var res = task.Result;
             }
         }
+
+        private string? GetTileKey(int z, int x, int y)
+        {
+            string keyPrefix = this._pathUtils.GetTilePathWithoutExtension(this.path, z, x, y, true);
+            var listRequests = new ListObjectsV2Request { BucketName = this._bucket, Prefix = keyPrefix, MaxKeys = 1 };
+
+            var listObjectsTask = this._client.ListObjectsV2Async(listRequests);
+            return listObjectsTask.Result.S3Objects.FirstOrDefault()?.Key;
```

Comment:
as we defined that source that contains files that are not tiles is invalid it its unnecessary as it the first (and should be only) file should be valid image or the entire source considered invalid.

as for the actual tile format validity, the format is later determined (in the overload used by the service) by  checking the binary data. this might be useful addition for the "getNextNBatch" function for cli use.

Reply:
opened bug